common/src/calc.rs

```diff
@@ -35,6 +38,44 @@ where
     }
 }
 
+/// Returns the number expressed as a string in exponential notation
+/// with the given precision (number of significant figures),
+/// optionally removing trailing zeros from the mantissa.
+///
+/// Examples:
+///
+/// ```text
+/// precision = 4, trim_end_zeroes = false
+///     1234124124 -> 1.234e9
+///     10000000 -> 1.000e7
+/// precision = 3, trim_end_zeroes = true
+///     1234124124 -> 1.23e9
+///     10000000 -> 1e7
+/// ```
+#[inline]
+pub fn to_exponential_notation<T>(value: T, precision: usize, trim_end_zeros: bool) -> String
+where
+    T: Into<U256> + Display,
```

Comment:
this will break if we eventually move to `ruint::U256`

can we get rid if the generic and accept U256 instead, this should also reduce generated code

Reply:
sure